packages/mask/src/components/shared/WalletStatusBox/TransactionList.tsx

```diff
@@ -0,0 +1,188 @@
+import { useAsync } from 'react-use'
+import Services from '../../../extension/service'
+import type { RecentTransaction } from '../../../plugins/Wallet/services/transaction'
+import { useI18N } from '../../../utils'
+
+const useStyles = makeStyles()((theme) => ({
+    list: {
+        borderRadius: 8,
+        overflow: 'hidden',
+    },
+    listItem: {
+        height: 54,
+        backgroundColor: theme.palette.background.paper,
+        '&:nth-child(even)': {
+            backgroundColor: theme.palette.background.default,
+        },
+    },
+    transaction: {
+        width: '100%',
+    },
+    methodName: {
+        fontWeight: 'bold',
+        textTransform: 'capitalize',
+    },
+    cell: {
+        display: 'flex',
+        alignItems: 'center',
+        padding: theme.spacing(0, 0.5),
+        whiteSpace: 'nowrap',
+        color: theme.palette.text.primary,
+        boxSizing: 'border-box',
+    },
+    link: {
+        fill: 'none',
+    },
+    linkIcon: {
+        fill: 'none',
+        width: 12,
+        height: 12,
+        marginLeft: theme.spacing(0.5),
+    },
+}))
+
+const statusTextColorMap: Record<TransactionStatusType, string> = {
+    [TransactionStatusType.NOT_DEPEND]: '#FFB915',
+    [TransactionStatusType.SUCCEED]: '#60DFAB',
+    [TransactionStatusType.FAILED]: '#FF5F5F',
+    [TransactionStatusType.CANCELLED]: '#FF5F5F',
+}
+const statusTextMap: Record<TransactionStatusType, string> = {
+    [TransactionStatusType.NOT_DEPEND]: 'Pending',
+    [TransactionStatusType.SUCCEED]: 'Success',
+    [TransactionStatusType.FAILED]: 'Failed',
+    [TransactionStatusType.CANCELLED]: 'Cancelled',
+}
+
+const getContractFunctionName = async (data: string | undefined) => {
+    if (!data) return null
+    const sig = data.slice(0, 10)
+    const name = await Services.Ethereum.getContractFunctionName(sig)
+    return name
+}
+
+interface TransactionProps extends GridProps {
+    chainId: ChainId
+    transaction: RecentTransaction
+    onClear?(tx: RecentTransaction): void
+}
+const Transaction: FC<TransactionProps> = ({ chainId, transaction: tx, onClear = noop, ...rest }) => {
+    const { t } = useI18N()
+    const { Utils } = useWeb3State()
+    const { classes, theme } = useStyles()
+
+    const web3 = useWeb3()
+    const { value: contractAddress } = useAsync(async () => {
+        if (tx.receipt?.contractAddress) return tx.receipt.contractAddress
+        if (tx.payload?.params?.[0].to) return tx.payload.params[0].to as string
+        const transaction = await web3.eth.getTransaction(tx.hash)
+        return transaction.to
+    }, [web3, tx])
+    const address = (contractAddress || '').toLowerCase()
+    const txData = tx.payload?.params?.[0].data as string | undefined
+    const { value: functionName } = useAsync(async () => {
+        const name = await getContractFunctionName(txData)
+        if (name === TransactionType.CREATE_RED_PACKET) return 'Create Lucky Drop'
```

Comment:
What about other `TrasnactionType`?

Reply:
It's naming migration.
I have rename the enum, no need to do such fixing anymore.